Docs/Book/Cookbook.rst

```diff
@@ -622,6 +653,369 @@ Returning Substructure Matches as SMILES
+                        subImgSize=(250,250), useSVG=False)
+
+
+.. image:: images/RDKitCB_21_im2.png
+
+.. testcode::
+
+   # Regioisomer Hashes (from slide 17 in [ref2])
+   # Find Regioisomer matches for this molecule
+   r0 = Chem.MolFromSmiles('CC1=CC2=C(C=C1)C(=CN2CCN1CCOCC1)C(=O)C1=CC=CC2=C1C=CC=C2')
+   r0
+
+.. image:: images/RDKitCB_21_im3.png
+
+.. testcode::
+
+   # Calculate the regioisomer hash for r0
+   r0_regioHash = rdMolHash.MolHash(r0,rdkit.Chem.rdMolHash.HashFunction.Regioisomer)
+   print(r0_regioHash)
+
+.. testoutput::
+
+   *C.*C(*)=O.*CC*.C1COCCN1.c1ccc2[nH]ccc2c1.c1ccc2ccccc2c1
+
+.. testcode::
+
+   r0_regioHash_mol = Chem.MolFromSmiles('*C.*C(*)=O.*CC*.C1COCCN1.c1ccc2[nH]ccc2c1.c1ccc2ccccc2c1')
+   r0_regioHash_mol
+
+.. image:: images/RDKitCB_21_im4.png
+
+.. testcode::
+
+   # Create a list of SMILES
+   rList = ['CC1=CC2=C(C=C1)C(=CN2CCN1CCOCC1)C(=O)C1=CC=CC2=C1C=CC=C2',\
+           'CCCCCN1C=C(C2=CC=CC=C21)C(=O)C3=CC=CC4=CC=CC=C43',\
+           'CC1COCCN1CCN1C=C(C(=O)C2=CC=CC3=C2C=CC=C3)C2=C1C=CC=C2',\
+            'CC1=CC=C(C(=O)C2=CN(CCN3CCOCC3)C3=C2C=CC=C3)C2=C1C=CC=C2',\
+           'CC1=C(CCN2CCOCC2)C2=C(C=CC=C2)N1C(=O)C1=CC=CC2=CC=CC=C12',\
+           'CN1CCN(C(C1)CN2C=C(C3=CC=CC=C32)C(=O)C4=CC=CC5=CC=CC=C54)C']
+   # Loop through the SMILES rList and create RDKit molecular objects
+   rMols = [Chem.MolFromSmiles(r) for r in rList]
+
+.. testcode::
+
+   # Calculate Regioisomer Hashes
+   regioHashList = [rdMolHash.MolHash(rMol, rdkit.Chem.rdMolHash.HashFunction.Regioisomer) for rMol in rMols]
+   print(regioHashList)
+
+.. testoutput::
+   :options: -ELLIPSIS, +NORMALIZE_WHITESPACE
+
+   ['*C.*C(*)=O.*CC*.C1COCCN1.c1ccc2[nH]ccc2c1.c1ccc2ccccc2c1',
+    '*C(*)=O.*CCCCC.c1ccc2[nH]ccc2c1.c1ccc2ccccc2c1',
+    '*C.*C(*)=O.*CC*.C1COCCN1.c1ccc2[nH]ccc2c1.c1ccc2ccccc2c1',
+    '*C.*C(*)=O.*CC*.C1COCCN1.c1ccc2[nH]ccc2c1.c1ccc2ccccc2c1',
+    '*C.*C(*)=O.*CC*.C1COCCN1.c1ccc2[nH]ccc2c1.c1ccc2ccccc2c1',
+    '*C.*C.*C(*)=O.*C*.C1CNCCN1.c1ccc2[nH]ccc2c1.c1ccc2ccccc2c1']
+
+.. testcode::
+
+   rmatches =[]
+   for regioHash in regioHashList:
+       if regioHash == r0_regioHash:
+           print('Regioisomer: True')
+           rmatches.append('Regioisomer: True')
+       else:
+           print('Regioisomer: False')
+           rmatches.append('Regioisomer: False')
+
+.. testoutput::
+
+   Regioisomer: True
+   Regioisomer: False
+   Regioisomer: True
+   Regioisomer: True
+   Regioisomer: True
+   Regioisomer: False
+
+.. testcode::
+
+   # Create some labels
+   index = ['r0: ','r1: ','r2: ','r3: ','r4: ','r5: ']
+   labelList = [rmatches + index for rmatches,index in zip(index,rmatches)]
+   # Display molecules with labels
+   Draw.MolsToGridImage(rMols,legends=[label for label in labelList],
+                       subImgSize=(250,250), useSVG=False)
+   # note, that r0 is the initial molecule we were interested in.
+
+.. image:: images/RDKitCB_21_im5.png
+
+Contiguous Rotable Bonds
```

Comment:
I like the recipe, but the example is kind of boring. How about adding an atom or two so that there's a path with more than one rotatable bond in it?

Would also be nice to show the molecule with the rotatable bond paths highlighted on it! :-)

Reply:
good point, new example is more interesting. + highlighting added.